core/order.go

```diff
@@ -169,9 +206,22 @@ func prepareModeratedOrderContract(data *PurchaseData, n *OpenBazaarNode, contra
 	if err != nil {
 		return nil, err
 	}
-	payment.Amount = total
+	defn, _ := repo.LoadCurrencyDefinitions().Lookup(data.PaymentCoin)
+	payment.AmountValue = &pb.CurrencyValue{
+		Currency: &pb.CurrencyDefinition{
+			Code:         data.PaymentCoin,
```

Comment:
Use `defn` for value in case we decide to normalize through this abstraction later.